lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+      Error.captureStackTrace(this, this.constructor);
+    } else {
+      this.stack = (new Error(message)).stack;
+    }
+  }
+}
+
+class TimeoutError extends ExtendableError {}
+
+class BCNLMessage {
+  constructor(bytes) {
+    if (new.target === BCNLMessage) {
+      throw new TypeError(`Cannot construct ${new.target.name} instances directly`);
+    }
+
+    this.payload = Buffer.from(bytes);
+  }
+
+  static get USB_BLOCKSIZE() {
+    return 64;
+  }
+  static get MAGIC_HEADER() {
+    return 'ABC';
+  }
+  static get READ_TIMEOUT_MS() {
+    return 4000;
+  }
+  static get SEND_DELAY_MS() {
+    return 500;
+  }
+
+  static get ASCII_CONTROL() {
+    return {
+      ACK: 0x06,
+      CR: 0x0D,
+      ENQ: 0x05,
+      EOT: 0x04,
+      ETB: 0x17,
+      ETX: 0x03,
+      LF: 0x0A,
+      NAK: 0x15,
+      STX: 0x02,
+    };
+  }
+
+  static wait(delay) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, delay);
+    });
+  }
+
+  toString() {
+    return this.payload.toString('hex');
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  readMessage(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS) {
+    return new Promise((resolve, reject) => {
+      const abortTimer = setTimeout(() => {
+        debug('TIMEOUT');
+        reject(new TimeoutError('Timeout error.'));
+      }, readTimeout);
+
+      let message = Buffer.alloc(0);
+
+      async.doWhilst(
+        (callback) => {
+          hidDevice.receive((raw) => {
+            const packet = Buffer.from(new Uint8Array(raw));
+
+            // Only process if we get data
+            if (packet.length === 0) {
+              return callback(false);
+            }
+
+            const header = packet.slice(0, 3).toString('ascii');
+            const size = packet[3];
+
+            if (header !== BCNLMessage.MAGIC_HEADER) {
+              debug('Invalid packet from Contour device');
+              clearTimeout(abortTimer);
+              return callback(new Error('Invalid USB packet received.'));
+            }
+
+            message = Buffer.concat([message, packet.slice(4)],
+              (message.length + packet.length) - 4);
+
+            // USB_BLOCKSIZE - 4, because we don't include the MAGIC_HEADER or the size byte
+            if (size < (BCNLMessage.USB_BLOCKSIZE - 4)) {
+              clearTimeout(abortTimer);
+              return callback(true);
+            }
+            return callback(false);
+          });
+        },
+        (valid) => {
+          if (valid instanceof Error) {
+            throw valid;
+          }
+          return (valid !== true);
+        },
+        () => resolve(message),
+      );
+    });
+  }
+
+  sendMessage(hidDevice) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      let pos = 0;
+      const message = this.payload;
+
+      async.doWhilst(
+        (callback) => {
```

Comment:
I was almost going to comment on the ES6 syntax as well - mostly it was the use of classes that caught my eye. In general I definitely am supportive of using ES6 in new code (though we have a policy in the front-end stuff of not mixing code - so if you're editing an old ES5 file don't sprinkle a bit of ES6 syntax in there even if you can...it's jarring & confusing to read like that). But in this case I feel like Gerrit's opinion should carry a lot of weight since you and him are going to be the ones going back and forth between ES5 and ES6 driver code, so maybe a bit of a discussion to be had when he gets back. (I suspect he won't be *violently* opposed, but that's about as far as I'm comfortable guessing at his opinion.)